src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -246,6 +293,53 @@ internal static class PeWriter
             var peBlob = new BlobBuilder();
             var peContentId = peBuilder.Serialize(peBlob, out Blob mvidSectionFixup);
 
+            // Remove this code when  https://github.com/dotnet/roslyn/issues/25185 is fixed
+            if (SRMVisibleMachine != properties.Machine)
+            {
+                // ARM64 pe
+                // Update Amd64 machine type in PE header to 0xAA64
+                // Machine type in PEHeader is located at...
+                // 128 bytes (dos header)
+                // 4 bytes (PE Signature)
+                // 2 bytes (Machine)
+                // 2 bytes (NumberOfSections)
+                // 4 bytes (TimeDateStamp)
+                int machineOffset = 128 + 4;
+                int timeDateStampOffset = 128 + 4 + 2 + 2;
```

Comment:
>timeDateStampOffset [](start = 20, length = 19)

I don't believe these are at fixed offsets. I common scenarios yes, this will work. But I don't believe we always write things in precisely this order with nothing else.

Reply:
System.Reflection.Metadata (at least the old version we are using here) will always put this here. I can add lfaNew processing to allow it to be in a different spot, but I doubt that S.R.M will ever change this part of the rule.

Reply:
Is that the only timestamp we write? I didn't think so.

---
In reply to: [172383076](https://github.com/dotnet/roslyn/pull/25186#discussion_r172383076) [](ancestors = 172383076)

Reply:
I don't know how to gain confidence from looking at this code that it is correct.

---
In reply to: [172383316](https://github.com/dotnet/roslyn/pull/25186#discussion_r172383316) [](ancestors = 172383316,172383076)